distributed/worker.py

```diff
@@ -3114,17 +3132,21 @@ async def gather_dep(
                 del data, response
                 self.transitions(recommendations, stimulus_id=stimulus_id)
 
-                if not busy:
-                    self.repetitively_busy = 0
-                else:
-                    # Exponential backoff to avoid hammering scheduler/worker
-                    self.repetitively_busy += 1
-                    await asyncio.sleep(0.100 * 1.5**self.repetitively_busy)
-
-                    await self.query_who_has(*to_gather_keys)
+                if refresh_who_has:
+                    # All workers that hold known replicas of our tasks are busy.
+                    # Try querying the scheduler for unknown ones.
+                    who_has = await retry_operation(
+                        self.scheduler.who_has, keys=refresh_who_has
+                    )
+                    self.update_who_has(who_has)
```

Comment:
Notably this query to the scheduler does not happen if all workers that are known to hold a replica are in flight. Not sure if this disparity of treatment is intentional?